RustFxA/Avatar.swift

```diff
@@ -6,31 +6,40 @@ import UIKit
 import Shared
 
 open class Avatar {
-    open var image = Deferred<UIImage>()
+    open var image : UIImage?
     public let url: URL?
 
     init(url: URL?) {
         self.url = url
-        downloadAvatar()
-    }
-
-    private func downloadAvatar() {
-        guard let url = url else { return }
-
-        DefaultImageLoadingHandler.shared.getImageFromCacheOrDownload(
-            with: url,
-            limit: ImageLoadingConstants.NoLimitImageSize
-        ) { image, error in
+        downloadAvatar { image, error in
             guard error == nil,
                   let image = image
             else {
-                self.image.fill(UIImage(named: ImageIdentifiers.placeholderAvatar)!)
+                self.image = UIImage(named: ImageIdentifiers.placeholderAvatar)
                 NotificationCenter.default.post(name: .FirefoxAccountProfileChanged, object: self)
                 return
             }
-
-            self.image.fill(image)
+            
+            self.image = image
             NotificationCenter.default.post(name: .FirefoxAccountProfileChanged, object: self)
         }
     }
+
+    private func downloadAvatar(completionHandler: @escaping(UIImage?, Error?) -> Void) {
+        guard let url = url else { return }
```

Comment:
When there's no image we default to the `placeholderAvatar`, so I think if there's no URL we should call the completion handler with an empty image and error. So when there's no URL, we have the place holder avatar as well.

Reply:
That's better bro. Is there any defined `Error` for it? I only found an enum, `ImageLoadingError.unableToFetchImage`.

```swift
    private func downloadAvatar(completionHandler: @escaping(UIImage?, Error?) -> Void) {
        guard let url = url else {
            completionHandler(nil, ImageLoadingError.unableToFetchImage)
            return
        }
```